### PR TITLE
cluster local test in pilot suite

### PR DESCRIPTION
**Pull request by stevenctl**
Another attempt of moving things out of the multicluster/ package.

### CODE REVIEW

File: tests/integration/pilot/multicluster_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+func patchMeshConfig(t framework.TestContext, clusters cluster.Clusters, patch string) {
+	errG := multierror.Group{}
+	origCfg := map[string]map[string]string{}
+	mu := sync.RWMutex{}
+	for _, c := range clusters.Kube() {
+		c := c
+		errG.Go(func() error {
+			cm, err := c.CoreV1().ConfigMaps(i.Settings().SystemNamespace).Get(context.TODO(), "istio", v1.GetOptions{})
```
Comment (howardjohn) [inline tests/integration/pilot/multicluster_test.go:79]: can you respect the `istio.test.revision` flag?

Comment (howardjohn) [reply tests/integration/pilot/multicluster_test.go:79]: to fetch `istio-<revision>`

**Review by howardjohn — COMMENTED** [review]
patching mesh config seems not ideal - its really slow at the very least, and may make it hard to run in external istiod.

Can we merge https://github.com/istio/istio/pull/31410 first at least? then it should be instant to patch it?

**Review by howardjohn — APPROVED** [review]
feel free to remove hold once https://github.com/istio/istio/pull/31410 lands

File: pkg/config/mesh/watcher.go [hunk 2]
```diff
@@ -110,8 +110,9 @@ func (w *InternalWatcher) HandleMeshConfig(meshConfig *meshconfig.MeshConfig) {
 	}
 	w.mutex.Unlock()
 
-	for _, h := range handlers {
-		h()
+	// TODO hack: the first handler added is the ConfigPush, other handlers affect what will be pushed, so reversing iteration
+	for i := len(handlers) - 1; i >= 0; i-- {
```
Comment (stevenctl) [inline pkg/config/mesh/watcher.go:114]: cc @nmittler

**Pull request merged**